vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/Checkbox.java

```diff
@@ -138,7 +128,7 @@ private Label appendLabelElement() {
      */
     @Override
     public void setLabel(String label) {
-        labelElement.setText(label);
+        getElement().setProperty("label", label == null ? "" : label);
```

Comment:
I'm guessing `getLabel()` wouldn't work after this change because it tries to find a `<label>` element from inside the component but now there's no server-side representation for it.

Reply:
Updated to use client-side property, same as for other components like `TextField`.